src/cljc/athens/parser.cljc

```diff
@@ -109,7 +109,9 @@
      :url-link-url-parts     (fn [& chars]
                                (string/join chars))
      :any-chars              (fn [& chars]
-                               (string/join chars))}
+                               (string/join chars))
+     :component              (fn [raw-content-string]
+                               (into [] (concat [:component raw-content-string] (into [] (rest (block-parser raw-content-string))))))}
```

Comment:
Really bad code smell using 2 `into`s and a `concat`. Without jumping into a REPL, I'm pretty confident you only need one of these vector creation functions, not 3.

Reply:
Solved!